docs/guides/typescript.md

```diff
@@ -253,4 +188,58 @@ type State =
 /* ... */
 ```
 
-:::
+## Config Objects
+
+Machine config objects can be typed. This is useful when defining a machine config object _outside_ of the `createMachine(...)` function, and helps prevent [inference errors](https://github.com/davidkpiano/xstate/issues/310). The first and third generic parameters for `MachineConfig<TContext, TSchema, TEvent>` are the same as the first and second parameters for `createMachine<TContext, TEvent, TTypeState>` respectively. The second generic parameter 'TSchema' accepts an object type with a single key 'states' which lists the possible states of machines accepting the configuration:
+
+```ts
+import { MachineConfig } from 'xstate';
+
+type myStateSchema = {
+  states: {
+    stopped: {};
+    started: {};
+  };
+};
+
+const myMachineConfig: MachineConfig<TContext, myStateSchema, TEvent> = {
+  id: 'controller',
+  initial: 'stopped',
+  states: {
+    stopped: {
+      /* ... */
+    },
+    started: {
+      /* ... */
+    }
+  }
+  // ...
+};
+```
+
+## Actions
+
+The `send` action on the interpreted machine `interpret(stateMachine)` isn't always type safe. To get typechecking for this function signature, use the following pattern:
+
+```ts
+type UserEvents = {
+  type: 'TEST';
+  value: string;
+};
+
+const service = interpret(stateMachine);
```

Comment:
I know this isn't new, but I think we should note how `UserEvents` is used. It's only clear if you already know how types work with XState.

```suggestion
// type is attached to `stateMachine` (not shown)
type UserEvents = {
  type: 'TEST';
  value: string;
};

// `service` is aware that `stateMachine` events are of type `UserEvents`
const service = interpret(stateMachine);
```

Reply:
Good point, I'll update the example

Reply:
Wait, I have. Check out https://github.com/davidkpiano/xstate/pull/2235 that builds on this PR